src/opendr/engine/data.py

```diff
@@ -283,8 +300,42 @@ def open(cls, filename):
             raise FileNotFoundError('The image file does not exist.')
         data = cv2.imread(filename)
         data = cv2.cvtColor(data, cv2.COLOR_BGR2RGB)
+        data = np.transpose(data, (2, 0, 1))
         return cls(data)
 
+    def opencv(self):
+        """
+        Returns the stored image into a format that can be directly used by OpenCV.
+        This function is useful due to the discrepancy between the way images are stored:
+        HWC/BGR (OpenCV) and CWH/RGB (OpenDR/PyTorch)
+        :return: an image into OpenCV compliant-format
+        :rtype: NumPy array
+        """
+        data = np.transpose(self.data, (1, 2, 0))
+        data = cv2.cvtColor(data, cv2.COLOR_RGB2BGR)
+        return data
+
+    def convert(self, format='channels_first', channel_order='rgb'):
+        """
+        Returns the data in channels first/last format using either 'rgb' or 'bgr' ordering
+        :param format: either 'channels_first' or 'channels_last'
+        :param channel_order: either 'rgb' or 'bgr'
+        :return an image with the appropriate format
+        :rtype NumPy array
+        """
+        data = self.data
```

Comment:
This has to be a copy instead of a reference!

Reply:
I would like to avoid as much as possible copy operations on this function that is envisioned to be used by OpenDR implementations to easily change between different formats to avoid double conversions just to follow OpenDR format. Users are supposed to use `numpy()` to get a reference to internal data and `opencv()` to get a copy for visualization. So, indeed, this function may or may not copy the memory depending on the situation. Perhaps we should explicitly state this to avoid cases where users are not aware of this or even hide this function from the external Image interface.

Reply:
I realize now that it is okay since both `np.transpose()` and `cv2.cvtColor()` return copies instead of references. Thus, we do not have to create an additional copy in this line if we just want to avoid changing the internal `data` representation. However, it is not so nice to change the returning behavior based on the input arguments without being intended by the user.

Strictly speaking, `numpy()` should rather return a copy as well since the `data` property might not be a numpy array for the generic `Data` class. Hence, for consistency with those inherited types, which do not use an internal numpy class for their `data` property, the `numpy()` method of the `Image` type should probably rather return a copy as well.

Reply:
I fully agree and - indeed - this is a way to introduce nasty bugs for the end users... So, for `numpy()` we should probably return a copy (I will update it), since it is tailored to the end users. My only concern for `convert()` is that we add an additional overhead even for cases where `convert()` wouldn't do any conversion. Since `convert()` is used by various Learners this will probably add an overhead (which - I do not know if it really matters). We could force `convert()` to always copy the data just to be on the safe side.

Reply:
Maybe, we should change the `numpy()` method in a separate PR as this might not only affect `Image` (most probably).

Some options that I can think of:
1) State it in the documentation.
2) Always return a copy.
3) Add a parameter `enforce_copy` to always return a copy or use the current implementation, i.e., return a reference if converting is not required.

Reply:
Sounds good.